### PR TITLE
replaced automake with cmake and kytocabinet with leveldb

### PR DESCRIPTION
**Pull request by acgreek**
actually almost completely overhauled everything. Feel free to reject my pull request if I went to far from what you want, I'll ask github admins to disconnect our forks.

### CODE REVIEW

**Comment by 2ion** [comment]
Looks great! Not a fan of cmake myself, but leveldb sure is interesting.

File: queue.c [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+static const char* CmpName(void* arg) {
+	return "foo";
+
+}
+
+int queue_is_opened (const struct Queue * const q) {
+	return NULL != q->db;
+}
+
+struct Queue * queue_open_with_options(const char * const path,... ) {
```
Comment (2ion) [inline queue.c]: Why not use ORed flags though?
```
option1 | option2 | option3
```
That would look a lot cleaner and also save string comparisons.

Comment (2ion) [inline queue.c]: And also get rid of the vararg.

Comment (acgreek) [inline queue.c]: Some options take argument such as 'block size' etc. What do you have against vararg?

**Pull request merged**